quaddicted-random-map.py

```diff
@@ -52,17 +53,20 @@ def __init__(self) -> None:
         # by default, where this python file resides
         self.execution_path = dirname(abspath(__file__))
 
-    @classmethod
-    def check_quake_folder(cls) -> None:
-        if not os.path.exists(cls.QUAKE_MAPS_PATH):
-            print("> ERROR can't find Quake maps folder ('{}')".format(cls.QUAKE_MAPS_PATH))
+    def check_quake_folder(self) -> None:
+        if not os.path.exists(self.maps_path):
+            print("> ERROR can't find Quake maps folder ('{}')".format(self.maps_path))
             exit(1)
 
     def set_engine_binary(self, engine_binary: str) -> None:
         self.engine_binary = engine_binary.lower()
 
     def set_execution_path(self, execution_path: str) -> None:
-        self.execution_path = execution_path
+        self.execution_path = execution_path.replace(".", dirname(abspath(__file__)))
```

Comment:
what if you have a folder with a `.` in the name?
I think it'd be safer to do something like `.rfind(".")` + a replace so only changes the last occurrence, or something similar, don't you think?

Reply:
totally! I actually realized `os.path.abspath` would do the job nicely as well, basically assumes `.` is current dir, updated in last commit